### PR TITLE
configs: create filewatches during tiltfile creation

### PR DESCRIPTION
**Pull request by nicks**
Hello @milas,

Please review the following commits I made in branch nicks/fw:

9af8f1d2de8b363c8f9e20e9a0763640295c88cb (2021-06-28 11:55:02 -0400)
configs: create filewatches during tiltfile creation

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by nicks** [comment]
this is mainly taking the code in manifest subscriber and moving it over to tiltfile creation...

**Comment by nicks** [comment]
this should be much simpler and easier to reason about in general because:
1) the tiltfile loader already reconciles against the apiserver, handling garbage collection, etc, and
2) we really only want to change these filewatches based on the tiltfile, nothing else

File: internal/store/engine_state.go [hunk 1]
```diff
@@ -492,7 +491,6 @@ func NewState() *EngineState {
 	}
 
 	ret.Cmds = make(map[string]*Cmd)
-	ret.FileWatches = make(map[types.NamespacedName]*v1alpha1.FileWatch)
```
Comment (milas) [inline internal/store/engine_state.go:495]: It's so nice to see these go away

**Review by milas — APPROVED** [review]

File: internal/store/engine_state.go (hunk 1 above)
Comment (nicks) [reply internal/store/engine_state.go:495]: :clap:

**Pull request merged**